File: archives/C#/0110. Balanced Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsBalanced(TreeNode root) {
        if (root==null){
            return true;
        }
        return IsBalanced(root.left) && IsBalanced(root.right) &&(Math.Abs(GetHeight(root.left)-GetHeight(root.right))<=1);
    }
    
    public int GetHeight(TreeNode root){
        if (root==null){
            return 0;
        }
        return 1+Math.Max(GetHeight(root.right),GetHeight(root.left));
    }
}

File: archives/20190519python/0110. Balanced Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if not root:
            return True
        return self.isBalanced(root.left) and self.isBalanced(root.right) and abs(self.getHeight(root.left)-self.getHeight(root.right))<=1
    
    def getHeight(self,root):
        if not root:
            return 0
        return 1+max(self.getHeight(root.left),self.getHeight(root.right))
